src/core-services/product/mutations/updateProductsVisibility.js

```diff
@@ -0,0 +1,46 @@
+import executeBulkOperation from "../utils/executeBulkOperation.js";
+
+/**
+ *
+ * @method updateProductsVisibility
+ * @summary Updates a set a products visibility
+ * @param {Object} context -  an object containing the per-request state
+ * @param {Object} input - Input arguments for the bulk operation
+ * @param {String[]} input.productIds - an array of Product IDs
+ * @param {String} input.shopId - the shop's id
+ * @param {String[]} input.isVisible - the desired visibility
+ * @return {Boolean} A boolean flag that indicates whether the operation completed successfully or not.
+ */
+export default async function addTagsToProducts(context, input) {
+  const { productIds, shopId, isVisible } = input;
+  const { collections: { Products } } = context;
+  const totalProducts = productIds.length;
+  let success = false;
+
+  for (const _id of productIds) {
+    // TODO(pod-auth): figure out a better way to loop through this
+    // eslint-disable-next-line no-await-in-loop
+    await context.validatePermissions(`reaction:legacy:products:${_id}`, "update", { shopId });
+  }
+
+  // // Generate update statements
+  const operations = productIds.map((productId) => ({
+    updateOne: {
+      filter: {
+        _id: productId,
+        shopId
+      },
+      update: {
+        $set: {
+          isVisible
+        }
+      }
+    }
+  }));
+
+  const results = await executeBulkOperation(Products, operations, totalProducts);
+
+  success = totalProducts === results.updatedCount;
+
+  return success;
```

Comment:
This causes "success" that doesn't actually reflect whether or not the operation was successful. The updated count only counts things that could be updated. If you set the visibility to `false` and it's already `false` on one or more products, then `success` will `false` because not all of the products were updated.

I think success is determined whether or not the bulk operation itself ran. Perhaps reporting on how may documents were updated would work better.

Reply:
I agree, now returning `updatedCount` to the client.